system/src/Grav/Common/Assets/BaseAsset.php

```diff
@@ -171,6 +172,31 @@ public function setPosition($position)
 
         return $this;
     }
+    
+    /**
+     * Receive asset location and return the SRI integrity hash
+     * 
+     * @param $input
+     *
+     * @return string
+     */
+    public static function integrityHash( $input )
+    {
+        $grav = Grav::instance();
+
+        $assetsConfig = $grav['config']->get('system.assets');
+
+        if ( !empty($assetsConfig['enable_asset_sri']) && $assetsConfig['enable_asset_sri'] )
+        {
+            $dataToHash = file_get_contents( GRAV_ROOT . $input);
```

Comment:
@w00fz I think we should cache this part and maybe just check file modification time when hash recalculation isn't needed.

Reply:
i agree

Reply:
Yeah I agree, also it should probably also use the Grav File instance and locator for the asset content and path, rather than file_get_contents

Reply:
We can probably merge then optimized to be honest.